mgmt/rest/client/client.go

```diff
@@ -118,7 +124,7 @@ func (c *Client) do(method, path string, ct contentType, body ...[]byte) (*rbody
 			if strings.Contains(err.Error(), "tls: oversized record") || strings.Contains(err.Error(), "malformed HTTP response") {
 				return nil, fmt.Errorf("error connecting to API URI: %s. Do you have an http/https mismatch?", c.URL)
 			}
-			return nil, err
+			return nil, fmt.Errorf("URL target is not available. %v", err)
```

Comment:
It was to address the "throw a better error" and previous comment about valid target. It didn't make sense to do another http request to check if the target was valid when it will error if it isn't anyway. What extra changes were there?